### PR TITLE
google analytics measurement protocol

### PR DESCRIPTION
**Pull request by zhenz**
Add a hook to send messages to GA for tracking the downloads of complex appliances.

### CODE REVIEW

File: appliance_catalog/views.py [hunk 1]
```diff
@@ -141,8 +143,24 @@ def get_appliance_template(request, pk):
     logger.info('Getting and displaying YAML template for appliance')
 
     appliance = Appliance.objects.filter(pk=pk).get()
-    return HttpResponse(appliance.template, content_type="text/yaml")
 
+    # send message to GA
+    if settings.GOOGLE_ANALYTICS_PROPERTY_ID:
+        if request.user and request.user.username:
```
Comment (diurnalist) [inline appliance_catalog/views.py]: Do we have to attach the user's username to this? I'm not sure it's necessary, we can probably just use the random ID and not have to worry about storing things tied to user PII (personally identifiable information).

File: appliance_catalog/views.py [hunk 2]
```diff
@@ -25,6 +25,8 @@
 from keystoneclient.v3 import client
 from glanceclient import Client
 from smtplib import SMTPException
+from six.moves import http_client, urllib
```
Comment (diurnalist) [inline appliance_catalog/views.py]: Why this and not `requests`? Is there a standard one we use? @xandroc

File: appliance_catalog/views.py [hunk 3]
```diff
@@ -141,8 +143,24 @@ def get_appliance_template(request, pk):
...
+        params = urllib.parse.urlencode({'v': 1,
+                                         'tid': settings.GOOGLE_ANALYTICS_PROPERTY_ID,
+                                         'cid': cid,
+                                         't': 'event',
+                                         'ec': 'heat_template',
+                                         'ea': 'download',
+                                         'el': '{}-{}'.format(pk, appliance.name),
+                                         'ev': 0})
+        conn = http_client.HTTPConnection('www.google-analytics.com')
+        conn.request('POST', '/collect', params)
```
Comment (diurnalist) [inline appliance_catalog/views.py]: I'd feel more comfortable with this inside of a try/catch... otherwise this could break the appliance catalog.

**Comment by zhenz** [comment]
@diurnalist GOOGLE_ANALYTICS_PROPERTY_ID is set originally for other GA trackings, and it 's only set for prod. 
https://github.com/ChameleonCloud/portal/blob/master/chameleon_env.sample#L34
https://github.com/ChameleonCloud/portal/blob/master/chameleon/settings.py#L499
https://github.com/ChameleonCloud/portal/blob/master/chameleon/templates/includes/ga.html#L1

File: appliance_catalog/views.py (hunk 1 above)
Comment (zhenz) [reply appliance_catalog/views.py]: For users, I thought GA would have some interesting statistics, such as downloads by users. But as far as I know, there aren't any statistics on events related to users. I can just use a random id as client id.

File: appliance_catalog/views.py (hunk 2 above)
Comment (zhenz) [reply appliance_catalog/views.py]: "six" is used somewhere else and already installed, but "requests" is not.

File: appliance_catalog/views.py (hunk 3 above)
Comment (zhenz) [reply appliance_catalog/views.py]: will add try/catch

**Pull request merged**